packages/preview/src/browser/preview-uri.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import URI from "@theia/core/lib/common/uri";
+
+export namespace PreviewUri {
+    export const id = 'code-editor-preview';
+    export const param = 'open-handler=' + id;
+    export function match(uri: URI): boolean {
+        return uri.query.indexOf(param) !== -1;
+    }
+    export function encode(uri: URI): URI {
+        if (match(uri)) {
+            return uri;
+        }
+        const query = [param, ...uri.query.split('&')].sort().join('&');
```

Comment:
why `sort`?

Reply:
if someone types `[lalal](foo?a=1&b=2)` and `[lalal](foo?b=2&a=1)` should we open the same preview?

Reply:
good question. I think, that's an assumption about `foo?a=1&b=2` and `foo?b=2&a=1` being the same sub-resource. I don't think that true in general, e.g. `report?sortBy=A&sortBy=B` can be different to `report?sortBy=B&sortBy=A`.

Reply:
I will remove it

Reply:
It seems to be that trimming should be done by `Resource` not by `OpenHandler` since the latter does not really know.

Reply:
we should treat different queries as different URIs